### ISSUE TITLE
Array method type is not inferred

### ISSUE THREAD

**Issue by anvish** [Microsoft/TypeScript#14382]
**TypeScript Version:**  2.2.1

**Code**

```ts
const b: string[] | string[][] = []
b.map(x => 1)
// [ts] Cannot invoke an expression whose type lacks a call signature. 
// Type '{ <U>(this: [string, string, string, string, string], callbackfn: (value: string, index: number, ...' has no compatible call signatures.
```

**Expected behavior:**
Array.map type is detected 
**Actual behavior:**
Array.map type is not detected

**Comment by mhegazy** [comment]
Please see https://github.com/Microsoft/TypeScript/issues/10620
